.travis.yml

```
@@ -37,6 +37,8 @@ jobs:
     - env: TEST=code-coverage
 
 before_script:
+  - test -n "$DOCKER_USERNAME" && docker login -u "$DOCKER_USERNAME" --password "$DOCKER_PASSWORD" 
+  - test -n "$DOCKER_USERNAME" || echo "Docker Hub credentials unavailable, pulling anonymously"
```

If DOCKER_USERNAME is not defined, the `test` fails and thus makes Travis fail. Try this:
```suggestion
  - if [ -n "$DOCKER_USERNAME" ] ; then docker login -u "$DOCKER_USERNAME" --password ; else echo "Docker Hub credentials unavailable, pulling anonymously" ; fi
```